packages/realm-react/src/useObject.tsx

```diff
@@ -82,3 +152,24 @@ export function createUseObject(useRealm: () => Realm) {
     return new Proxy(object, {}) as T & Realm.Object<T>;
   };
 }
+
+// This is a helper function that determines if two primary keys are equal.  It will also handle the case where the primary key is an ObjectId or UUID
+const comparePrimaryKeys = (a: any, b: any): boolean => {
+  if (typeof a !== typeof b) {
+    return false;
+  }
+  if (typeof a === "string" || typeof a === "number") {
+    if (a === b) {
+      return true;
+    }
+  } else if (a instanceof Realm.BSON.ObjectId && b instanceof Realm.BSON.ObjectId) {
+    if (a.toHexString() === b.toHexString()) {
+      return true;
+    }
+  } else if (a instanceof Realm.BSON.UUID && b instanceof Realm.BSON.ObjectId) {
+    if (a.toHexString() === b.toHexString()) {
+      return true;
+    }
+  }
+  return false;
```

Comment:
An alternative. (And fixed a typo in the last if-block where it was `ObjectId` rather than `UUID`.)
```suggestion
  if (typeof a === "string" || typeof a === "number") {
    return a === b;
  }
  if (a instanceof Realm.BSON.ObjectId && b instanceof Realm.BSON.ObjectId) {
    return a.toHexString() === b.toHexString();
  }
  if (a instanceof Realm.BSON.UUID && b instanceof Realm.BSON.UUID) {
    return a.toHexString() === b.toHexString();
  }
  return false;
```